dotCMS/src/main/java/com/dotmarketing/cms/urlmap/URLMapAPIImpl.java

```diff
@@ -44,12 +40,11 @@
  */
 public class URLMapAPIImpl implements URLMapAPI {
 
-    private volatile Collection<ContentTypeURLPattern> patternsCache;
```

Comment:
why do you remove the 'volatile'?

Reply:
because the only method that writes to the patterns cache is synchronized?  It is ok if some threads get a dirty read as long as once it has been written, all threads will ultimately see the right variable.  Am I missing something?

Reply:
I think there might be a List implementation on Concurrent

Reply:
So the question is, do we need a concurrent implementation on an object when all writes happen in a synced block?  I don't think we do - though I am happy to be proven wrong.

Reply:
CopyOnWriteArrayList, might be

Reply:
Guys, I understand the better ways to create / use a concurrent List.  I am just not seeing where that is a requirement here.  Here are what the impl above expects

- Multiple threads read the List
- Any clearing/writing/modification to the List happens in 1 thread.   
- Dirty reads are ok for a few (milli) seconds as the list rebuilds

I have not read anything that says single threaded modification of an ArrayList is harmful.  All the examples show where there are multiple threads trying to write to a list or the ConcurrentMod exceptions use of looping over a list to modify.

I can change to a more defensive pattern, but I am not convinced it is broken.

Reply:
I think is ok